ai.c

```diff
@@ -1,7 +1,51 @@
 #include "ai.h"
+#include <math.h>
+#include <stdio.h>
+#include <stdlib.h>
 
-float getSpeedToSet(Vector distanceVector, float relativeBearing) {
-    return 0.0;
+float START_RADIUS = 9;
+float DEAD_ZONE_RADIUS = 3;
+
+/*
+Determines the speed percentage to be set by the AI.
+Inputs:
+distance - floating point representation of the distance to target in meters
+relativeBearing - floating point representation of the relative heading needed to get to the target in degrees
+
+Output:
+Overall speed percentage to be sent to the ESC
+*/
+
+
+float getSpeedToSet(float distance, float relativeBearing) {
+	//Determine speed to set
+	float absBearing = (float)(fabs(relativeBearing));
+	if(absBearing > 162.0){
```

Comment:
Should 162.0 be a defined constant?

Reply:
Also, if you wanted, you could use the ternary operator here. Ex. `absBearing = (absBearing > 162.0) ? 162.0 : absBearing`. Its just syntax sugar but compresses the 3 lines to 1.